cmd/local-gadget/advise/seccomp-profile.go

```diff
@@ -0,0 +1,141 @@
+
+	"github.com/spf13/cobra"
+
+	commonadvise "github.com/inspektor-gadget/inspektor-gadget/cmd/common/advise"
+	commonutils "github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/local-gadget/utils"
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+	seccompAdviseTracer "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/advise/seccomp/tracer"
+	localgadgetmanager "github.com/inspektor-gadget/inspektor-gadget/pkg/local-gadget-manager"
+)
+
+const localGadgetSubKey = "local-gadget-key"
+
+func newSeccompProfileCmd() *cobra.Command {
+	var commonFlags utils.CommonFlags
+
+	runCmd := func(cmd *cobra.Command, args []string) error {
+		if commonFlags.Containername == "" {
+			return commonutils.WrapInErrMissingArgs("--containername / -c")
+		}
+
+		localGadgetManager, err := localgadgetmanager.NewManager(commonFlags.RuntimeConfigs)
+		if err != nil {
+			return commonutils.WrapInErrManagerInit(err)
+		}
+		defer localGadgetManager.Close()
+
+		tracer, err := seccompAdviseTracer.NewTracer()
+		if err != nil {
+			return fmt.Errorf("creating tracer: %w", err)
+		}
+		defer tracer.Close()
+
+		var mntns uint64
+		containerNotifier := make(chan error)
+
+		// Subscribe to container creation and termination events. Termination
+		// is used to generate a profiles when a container terminates. Creation
+		// is used to store the container mount namespace in case it wasn't
+		// already running when gadget was launched.
+		containers := localGadgetManager.ContainerCollection.Subscribe(
+			localGadgetSubKey,
+			containercollection.ContainerSelector{
+				Name: commonFlags.Containername,
+			},
+			func(event containercollection.PubSubEvent) {
+				switch event.Type {
+				case containercollection.EventTypeAddContainer:
+					if mntns != 0 {
+						containerNotifier <- fmt.Errorf("multiple containers with name %q",
+							commonFlags.Containername)
+						break
+					}
+
+					mntns = event.Container.Mntns
+				case containercollection.EventTypeRemoveContainer:
+					containerNotifier <- nil
+				}
+			},
+		)
+		defer localGadgetManager.ContainerCollection.Unsubscribe(localGadgetSubKey)
+
+		n := len(containers)
+		if n > 1 {
+			return fmt.Errorf("multiple containers with name %q",
+				commonFlags.Containername)
+		}
+
+		// If the container already exists, we store the mntns to generate the
+		// profile later. Otherwise, the mntns will be retrieved on the
+		// EventTypeAddContainer event.
+		if n == 1 {
+			mntns = containers[0].Mntns
+		}
+
+		stop := make(chan os.Signal, 1)
+		signal.Notify(stop, syscall.SIGINT, syscall.SIGTERM)
+
+		// The profile can be generated upon container termination or when user
+		// requests it through a signal.
+	WaitForEvent:
+		for {
+			select {
+			case <-stop:
+				break WaitForEvent
+			case err := <-containerNotifier:
+				if err != nil {
+					return err
+				}
+				break WaitForEvent
+			}
+		}
```

Comment:
I suppose something like works the same?

```suggestion
		select {
		case <-stop:
		case err := <-containerNotifier:
			if err != nil {
				return err
			}
		}
```